src/afa/tests-afa.cc

```diff
@@ -428,6 +428,137 @@ TEST_CASE("Mata::Afa antichain emptiness test")
 
 }
 
+TEST_CASE("Mata::Afa::construct() from IntermediateAut correct calls")
+{ // {{{
+    Afa aut;
+    Mata::IntermediateAut inter_aut;
+    StringToSymbolMap symbol_map;
+
+    SECTION("construct an empty automaton")
+    {
+        inter_aut.automaton_type = Mata::IntermediateAut::AFA;
+        aut = Mata::Afa::construct(inter_aut);
+        REQUIRE(true);
+    }
+
+    SECTION("construct a simple non-empty automaton accepting the empty word from intermediate automaton")
+    {
+        std::string file =
+                "@AFA-explicit\n"
+                "%States-enum p q r\n"
+                "%Alphabet-auto\n"
+                "%Initial p | q\n"
+                "%Final p | q\n";
+        const auto auts = Mata::IntermediateAut::parse_from_mf(parse_mf(file));
+        inter_aut = auts[0];
+
+        aut = construct(inter_aut);
+
+        REQUIRE(aut.initialstates.size() == 2);
+        REQUIRE(aut.finalstates.size() == 2);
+    }
+
+    SECTION("construct an automaton with more than one initial/final states from intermediate automaton")
+    {
+        std::string file =
+                "@AFA-explicit\n"
+                "%States-enum p q 3\n"
+                "%Alphabet-auto\n"
+                "%Initial p | q\n"
+                "%Final p & q & r\n";
```

Comment:
I am a little bit confused, is there any difference between using `Final p & q & r` and `Final p | q | r` in this context of final states? I did not notice the difference in the code since each state `p, q, r` is simply stored to the `finalstates` container within the `Afa Mata::Afa::construct` function (regardless of the separator). Or did you just want to demonstrate that both symbols `|, &` are supported and that they can be used as separators in the context of final states?

Reply:
Yes, actually this is just for demonstration of syntactical abilities of MATA format. In your representation there is no difference between these two. IMHO there is a difference between definition of final states with `&` and `|` in theory of AFA.

Reply:
Ok, thanks for your answer.